c/src/detectors/c-null-pointer-dereference/c-null-pointer-dereference_compliant.c

```diff
@@ -0,0 +1,15 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+// {fact rule=c-null-pointer-dereference@v1.0 defects=0}
+#include <stdio.h>
+
+void compliant()
+{
+   // Compliant: Pointers are always initialized before use.
+   int *ptr = NULL;
```

Comment:
The `if` condition will always be `false`. I think a better way would be to accept `*ptr` as an argument